cclib/io/filereader.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Generic file reader and related tools"""
+
+
+class Reader(object):
+    """Abstract class for reader objects."""
```

Comment:
Do we really want it to be abstract? If so, let's use `abc`.

Reply:
I think that should be in a separate PR, since it needs to be done for the writers as well, along with tests for each.